File: SET3-Backend/Models/UserOrderModel.cs
using System.ComponentModel.DataAnnotations;

namespace SET3_Backend.Models
{
    public class UserOrderModel
    {
        [Key]
        public int Id { get; set; }
        public DateTime UpdatedDate { get; set; }
        public bool Done { get; set; }
        public int ShopId { get; set; }
        public int UserId { get; set; }
        public int CashRegisterId { get; set; }

        public UserOrderModel()
        {
        }

        public UserOrderModel(DateTime updatedDate, bool done, int shopId, int userId, int cashRegisterId)
        {
            UpdatedDate = updatedDate;
            Done = done;
            ShopId = shopId;
            UserId = userId;
            CashRegisterId = cashRegisterId;
        }
    }
}

File: lib/UI/components/orders/userordermodel.dart
import 'dart:core';

class UserOrderModel {
  int Id = 0;
  String UpdatedDate = "";
  bool Done = false;
  int ShopId = 0;
  int UserId = 0;
  int CashRegisterId = 0;

  UserOrderModel(
      this.Id, this.UpdatedDate, this.Done, this.ShopId, this.UserId, this.CashRegisterId);

  int get id => Id;
  String get date => UpdatedDate;
  bool get done => Done;
  int get shopID => ShopId;
  int get userID => UserId;
  int get cashRegisterID => CashRegisterId;

  factory UserOrderModel.fromJson(dynamic json) {
    return UserOrderModel(json['id'] as int, json['updatedDate'] as String,
        json['done'] as bool, json['shopId'] as int, json['userId'] as int, json['cashRegisterId']);
  }
  Map<String, dynamic> toMap() {
    var map = Map<String, dynamic>();
    map["id"] = Id;
    map["date"] = UpdatedDate;
    map["done"] = Done;
    map["shopID"] = ShopId;
    map["userID"] = UserId;
    map['cashRegisterID'] = CashRegisterId;
    return map;
  }

  UserOrderModel.fromObject(dynamic o) {
    this.Id = o["Id"];
    this.UpdatedDate = o["UpdatedDate"];
    this.Done = o["Done"];
    this.ShopId = o["ShopId"];
    this.UserId = o["UserId"];
    this.CashRegisterId = o["CashRegisterId"];
  }
}
